client/src/pages/dashboard/stores/DeferredStore.ts

```diff
@@ -0,0 +1,34 @@
+import { asyncWritable, readable } from '@square/svelte-store';
+import { DeferredRegistrationSchema, DeferredSnapshotSchema } from '../../../../../model/src/api.ts';
+import localForage from 'localforage';
+import { DeferredFetch } from '../../syncman.ts';
+import { assert } from '../../../assert.ts';
+import { Status } from '../../../../../model/src/snapshot.ts';
+import { z } from 'zod';
+import { topToastMessage } from './ToastStore.ts';
+
+export const latestMessage = readable(null as string | null, set=>{
+    function handle(evt: MessageEvent) {
+        set(z.string().parse(evt.data));
+        topToastMessage.enqueue({ title: 'Background Syncronization', body: 'Syncronization successful.' });
+    }
+    addEventListener('message', handle);
+    return () => removeEventListener('message', handle);
+});
+
+export const deferredSnaps = asyncWritable(
+    [],
+    async() => {
+        // Get all keys in the localStorage and resolve all of them and set as contents of this store.
+        const deferred = (await localForage.keys()).map(async key => {
+            const defer: DeferredFetch | null = await localForage.getItem(key);
```

Comment:
Let's assert the data type using Zod schemas here rather than "trusting" the type annotation. Recall that type annotations are just a compile-time construct. We still need to verify the types at runtime, especially considering that `localStorage` may easily be tampered with.